widget/table.go

```diff
@@ -0,0 +1,328 @@
+}
+
+type tableCells struct {
+	BaseWidget
+	t        *Table
+	cellSize fyne.Size
+}
+
+func newTableCells(t *Table, s fyne.Size) *tableCells {
+	c := &tableCells{t: t, cellSize: s}
+	c.ExtendBaseWidget(c)
+	return c
+}
+
+func (c *tableCells) CreateRenderer() fyne.WidgetRenderer {
+	return &tableCellsRenderer{cells: c, pool: &syncPool{}, objects: []fyne.CanvasObject{}, visible: make(map[cellID]fyne.CanvasObject)}
+}
+
+func (c *tableCells) Tapped(e *fyne.PointEvent) {
+	if e.Position.X < 0 || e.Position.X >= c.Size().Width || e.Position.Y < 0 || e.Position.Y >= c.Size().Height {
+		c.t.SelectedColumn = -1
+		c.t.SelectedRow = -1
+		c.t.Refresh()
+		return
+	}
+
+	c.t.SelectedColumn = e.Position.X / (c.cellSize.Width + tableDividerThickness)
+	c.t.SelectedRow = e.Position.Y / (c.cellSize.Height + tableDividerThickness)
+
+	if c.t.OnCellSelected != nil {
+		c.t.OnCellSelected(c.t.SelectedRow, c.t.SelectedColumn)
+	}
+
+	if c.t.moveCallback != nil {
+		c.t.moveCallback()
+	}
+}
+
+type cellID struct {
+	row, col int
+}
+
+type tableCellsRenderer struct {
+	cells   *tableCells
+	pool    pool
+	objects []fyne.CanvasObject
+	visible map[cellID]fyne.CanvasObject
+}
+
+func (r *tableCellsRenderer) Layout(_ fyne.Size) {
+	// we deal with cached objects so just refresh instead
+}
+
+func (r *tableCellsRenderer) MinSize() fyne.Size {
+	rows, cols := r.cells.t.DataSize()
+	return fyne.NewSize(r.cells.cellSize.Width*cols+(cols-1), r.cells.cellSize.Height*rows+(rows-1))
+}
+
+func (r *tableCellsRenderer) Refresh() {
+	template := r.cells.t.NewCell() // don't use cache, we need new template
+	oldSize := r.cells.cellSize
+	r.cells.cellSize = template.MinSize().Add(fyne.NewSize(theme.Padding()*2, theme.Padding()*2))
+	if oldSize != r.cells.cellSize { // theme changed probably
+		r.returnAllToPool()
+	}
+
+	rows, cols := r.visibleCount()
+	offX := r.cells.t.offset.X - (r.cells.t.offset.X % (r.cells.cellSize.Width + tableDividerThickness))
+	minCol := offX / (r.cells.cellSize.Width + tableDividerThickness)
+	offY := r.cells.t.offset.Y - (r.cells.t.offset.Y % (r.cells.cellSize.Height + tableDividerThickness))
+	minRow := offY / (r.cells.cellSize.Height + tableDividerThickness)
+
+	wasVisible := r.visible
+	r.visible = make(map[cellID]fyne.CanvasObject)
+	var cells []fyne.CanvasObject
+	for y := minRow; y < minRow+rows; y++ {
+		for x := minCol; x < minCol+cols; x++ {
+			id := cellID{y, x}
+			c, ok := wasVisible[id]
+			if !ok {
+				c = r.pool.Obtain()
+				if c == nil {
+					c = r.cells.t.NewCell()
+					c.Resize(r.cells.cellSize)
+				}
+
+				c.Move(fyne.NewPos(theme.Padding()+x*r.cells.cellSize.Width+(x-1)*tableDividerThickness,
+					theme.Padding()+y*r.cells.cellSize.Height+(y-1)*tableDividerThickness))
+
+				r.cells.t.UpdateCell(minRow+y, minCol+x, c)
+			}
+
+			r.visible[id] = c
+			cells = append(cells, c)
+		}
+	}
+
+	for id, old := range wasVisible {
+		if _, ok := r.visible[id]; !ok {
+			r.pool.Release(old)
+		}
+	}
+	r.objects = cells
```

Comment:
A nice sanity check just while this PR is in development would be add `log.Println(len(cells))` here to ensure only the minimum cells are rendered

Reply:
Yes I see you added those